bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -315,3 +293,58 @@ def x_more_than_y(x):
     
     func_call = jitter.jit_function(func, col_list, extra_args)
     return rdf._OriginalFilter("Numba::" + func_call, filter_name)
+
+def _PyDefine(rdf, col_name, callable_or_str, cols = [] , extra_args = {} ):
+    """
+    Defines a new column in the RDataFrame.
+    Arguments:
+    1. col_name: The name of the new column to be defined
+    2. callable_or_str: The definition of the contents of the new columns
+        It can be either a python callable or a c style string.
+    3. cols: list of columns that the callable will receive as argument.
+                If not provided then it tries maps the name of the parameter to a column name of the RDF.
+    4. extra_args: non-columnar arguments to be passed to the callable.
+    Returns:
+        RDataFrame: rdf with new column defined
+    
+    Examples:
+    1. rdf.Define("x", lambda: np.random.rand())
+        Define using a python lambda
+    2. rdf.Define("x2", lambda x: x*x, ["y"])
+        Maps the function argument to column y.
+    3. scale = 0.5
+        def x_scaled(x):
+            return x*scale
+       rdf.Define("X_scaled", x_scaled)
+       The scale is captured from scope and treated as a compile time constant during first jit.
+       Note: Any changes to scale will not be reflected in the jitted function as jitting is a one time operation.
+    4. def x_scaled(x, u):
+            return x*u
+       rdf.Define("x_scaled", x_scaled, extra_args = {"u":0.5})
+
+    """
+    if not isinstance(col_name, str):
+        raise TypeError(f"First argument of Define must be a valid string for the new column name. {type(col_name).__name__} is not a string.")
+
+    if isinstance(callable_or_str, str): # If string argument is passed. Invoke the Original Define.
+        return rdf._OriginalDefine(col_name, callable_or_str)
+
+    if not callable(callable_or_str): # The 2st argument is either a string or a python callable.
+        raise TypeError(f"The second argument of a Define operation should be a callable. {type(callable_or_str).__name__} object is not callable.")
+    
+    if not isinstance(cols, list):        
+        raise TypeError(f"Define takes a column list as third arguments but {type(cols).__name__} was given.")
+    
```

Comment:
This and the previous comment on cols being any iterable. 
They are not specific to the Defines. The same need to be done for the Filters and DefinePerSample
So, I'll put them in a separate PR, if that makes sense.
Otherwise, we would have modifications to _PyFilter in a PR that adds _PyDefine functionality.